moveit_ros/visualization/planning_scene_rviz_plugin/src/planning_scene_display.cpp

```diff
@@ -661,4 +665,28 @@ void PlanningSceneDisplay::fixedFrameChanged()
   calculateOffsetPosition();
 }
 
+#ifdef ROS_KINETIC
+#include <boost/thread/mutex.hpp>
+
+/* Unfortunately, in Kinetic rviz doesn't provide access to its tf2 buffer.
+   Hence, we maintain single other tf2 buffer to be use by all MoveIt rviz plugin instances */
+
+// Return (singleton) tf2 Transform Buffer shared between all MoveIt display instances
+std::shared_ptr<tf2_ros::Buffer> PlanningSceneDisplay::getTF2BufferPtr()
```

Comment:
This is quite elegant, and nicely mirrors the [SharedStorage usage of a static tf2::Buffer](https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp#L43-L92). Good thinking and well done.